crates/key-server/src/server.rs

```diff
@@ -133,19 +116,30 @@ struct FetchKeyResponse {
 #[derive(Clone)]
 struct Server {
     sui_client: SuiClient,
-    network: Network,
     master_key: IbeMasterKey,
-    key_server_object_id: ObjectID,
     key_server_object_id_sig: MasterKeyPOP,
-    sdk_version_requirement: VersionReq,
+    options: KeyServerOptions,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct KeyServerOptions {
```

Comment:
let's make the main ones below optional, with defaults:
checkpoint_update_interval, rgp_update_interval, allowed_staleness, session_key_ttl_max, sdk_version_requirement
no one should change those in the normal setting